### PR TITLE
added support for mimetype

### PR DESCRIPTION
**Pull request by dbudworth**
Upgraded `cloud.google.com/go/ai` to v0.4.0 and regenerated the veneer to get the `model.ResponseMimeType` field plumbed through so we can generate json responses.

Addresses issue #83

### CODE REVIEW

File: genai/generativelanguagepb_veneer.gen.go [hunk 1]
```diff
@@ -698,26 +700,34 @@ type GenerationConfig struct {
...
-	// attribute of the `Model` returned from the `getModel` function.
+	// attribute of the `Model` returned from the `getModel` function. Empty
+	// `top_k` field in `Model` indicates the model doesn't apply top-k sampling
+	// and doesn't allow setting `top_k` on requests.
 	TopK *int32
+	// Optional. Output response mimetype of the generated candidate text.
+	// Supported mimetype:
+	// `text/plain`: (default) Text output.
+	// `application/json`: JSON response in the candidates.
+	ResponseMimeType string
```
Comment (jba) [inline genai/generativelanguagepb_veneer.gen.go:716]: Spell as `ResponseMIMEType`. We like to have acronyms and other initialisms use the same case.

**Review by jba — CHANGES_REQUESTED** [review]
Thanks! One minor change.

File: genai/generativelanguagepb_veneer.gen.go (hunk 1 above)
Comment (dbudworth) [reply genai/generativelanguagepb_veneer.gen.go]: updated, by just hand editing the generated file.
All I could find on protoveneer, as far as docs, was the config.go from that project and it didn't seem to support field renaming.
Happy to change the config if y'all could point me towards docs on how.

Comment (jba) [reply genai/generativelanguagepb_veneer.gen.go]: Sorry, I should have told you how. 
There are no docs for protoveneer except in the code. :(
Here is a similar case in the config: https://github.com/google/generative-ai-go/blob/main/genai/config.yaml#L67

Comment (dbudworth) [reply genai/generativelanguagepb_veneer.gen.go]: nevermind, spoke too soon, figured out how the config works.  Just pushed a more proper fix

**Review by jba — APPROVED** [review]
Thanks!

**Pull request merged**